e2e/nomostest/prometheus_metrics.go

```diff
@@ -47,23 +47,29 @@ type MetricsPredicate func(context.Context, prometheusv1.API) error
 // until successful (with timeout).
 func ValidateMetrics(nt *NT, predicates ...MetricsPredicate) error {
 	ctx, cancel := context.WithCancel(nt.Context)
-	defer cancel() // stop port-forward
-	address, err := portForwardPrometheus(ctx, nt)
-	if err != nil {
-		return err
-	}
-	client, err := prometheusapi.NewClient(prometheusapi.Config{
-		Address: fmt.Sprintf("http://%s", address),
-	})
-	if err != nil {
-		return err
-	}
-	v1api := prometheusv1.NewAPI(client)
+	defer cancel()
+	var address string
+	var v1api prometheusv1.API
 
 	nt.T.Log("[METRICS] validating prometheus metrics...")
 	for i, predicate := range predicates {
 		duration, err := retry.Retry(nt.DefaultMetricsTimeout, func() error {
-			err := predicate(ctx, v1api)
+			port, err := nt.prometheusPortForwarder.GetPort()
+			if err != nil {
+				return err
+			}
+			newAddress := fmt.Sprintf("localhost:%d", port)
+			if newAddress != address {
```

Comment:
I have a feeling this is an unnecessary optimization that only really saves on a little memory churn from re-creating the client, but it's ok to leave it in. Maybe leave a comment to explain the added complexity tho.

Reply:
done